### PR TITLE
Scuemata: Add grafana-cli command to validate basic scuemata

### PR DESCRIPTION
**Pull request by dsotirakis**
**What this PR does / why we need it**:

This PR is responsible to add a `grafana-cli` command to validate Scuemata schemas.

**Which issue(s) this PR fixes**:

Fixes a checkbox in #33139

### CODE REVIEW

**Review by sdboyer — CHANGES_REQUESTED** [review]

File: pkg/cmd/grafana-cli/commands/cue_scuemata_validation_command.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/grafana/grafana"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/utils"
+	"github.com/grafana/grafana/pkg/schema"
+	"github.com/grafana/grafana/pkg/schema/load"
+)
+
+var p = load.BaseLoadPaths{
+	BaseCueFS:       grafana.CoreSchema,
+	DistPluginCueFS: grafana.PluginSchema,
+}
```
[the remark this answers is not in the document]
Comment (dsotirakis) [reply pkg/cmd/grafana-cli/commands/cue_scuemata_validation_command.go:14]: Yep - will do! (was off Friday - Monday)

**Comment by dsotirakis** [comment]
@sdboyer -- added some basic testing, I think the way it is now it's a bit difficult to test basic validation without changing the filesystem. However if we do change it, then tests will fail due to some hardcoded values [such as this](https://github.com/grafana/grafana/blob/master/pkg/schema/load/dashboard.go#L40). Creating a temp FS dir may work for other validation funcs, such as `SearchAndValidate` which I am not 100% sure if needs to be part of this cli command?

Also - was actually thinking about making `BaseLoadPaths` kind of "private" to the package so it can be get with a `getter/loader` and not be modified at all?

**Comment by sdboyer** [comment]
> I think the way it is now it's a bit difficult to test basic validation without changing the filesystem. However if we do change it, then tests will fail due to some hardcoded values such as this

Even with that hardcoded path, this should still be possible:

1. Create a `testdata` dir that's a sibling to the test file
2. Within that testdata dir, create a `cue/data/gen.cue` (or create multiple instances of that structure, one per test case). You'll probably also need to either copy over the other contents of `grafana/cue*` - or create some kind of quick hacky thing that lets you overlay the testdata dir onto the `grafana.CoreSchema` type. [This](https://github.com/laher/mergefs) might do it.
3. In `TestValidateScuemataBasics()`, call `os.DirFS("testdata")`, and do the overlay thing if that's the approach you take, and use that for `BaseLoadPaths.BaseCueFS` instead of `grafana.CoreSchema`.

> `SearchAndValidate` which I am not 100% sure if needs to be part of this cli command?

Yeah, that definitely doesn't - that's about taking an object (e.g. a dashboard) and finding which schema in a family it validates against (if any). That is, checking the correctness of resources wrt schema. What we're doing here is prerequisite to those checks making sense: we're checking the correctness of the schema wrt general scuemata invariants - backwards compat is followed within a lineage, there are no CUE syntax errors, etc.

File: pkg/cmd/grafana-cli/commands/scuemata_validation_command_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		}
+
+		err := validate(nil, baseLoadPaths, load.BaseDashboardFamily)
+		require.NoError(t, err, "error while loading base dashboard scuemata")
+
+		err = validate(nil, baseLoadPaths, load.DistDashboardFamily)
+		assert.EqualError(t, err, "all schema should be valid with respect to basic CUE rules, Family.lineages.0.0: field #Panel not allowed")
+	})
+
+	t.Run("Testing scuemata validity against JSON files", func(t *testing.T) {
```
Comment (sdboyer) [inline pkg/cmd/grafana-cli/commands/scuemata_validation_command_test.go:62]: I started a framework for testing dashboard JSON [here](https://github.com/grafana/grafana/blob/1b8c0ce88bdc3cc22564edb535b81863799b68c5/pkg/schema/load/load_test.go#L50) (idk why i didn't point you to this earlier, sorry!). This subtest appears to duplicate that, albeit with both valids and invalids. So, two comments there:

1. Let's keep this PR only to checking the validity of scuemata themselves, in keeping with the other comment about having a separate command for checking resources
2. While i'm not wedded to having the test i wrote live where it does today, there almost certainly should be only one framework for validating test input JSON against schema

(also, nit: The title of this test seems to flow in the wrong direction. Scuemata/schema can determine whether JSON/resources are valid, not the other way around.)

Comment (sdboyer) [reply pkg/cmd/grafana-cli/commands/scuemata_validation_command_test.go:62]: > Scuemata/schema can determine whether JSON/resources are valid, not the other way around.

Though, i should maybe clarify: _right now_, the JSON generated by the frontend application does not pass our schemas. That's a temporary situation that we absolutely have to fix in the coming weeks. Once we've fixed it, it should never occur again (barring bugs).

**Review by sdboyer — CHANGES_REQUESTED** [review]
Let's defer anything related to actually validating a resource until a subsequent PR. Really just need the scuemata invariant validation in this PR.

And let's try the overlay fs lib i linked to. I'd rather not copy these files if we can avoid it - it'll create toilsome, hard-to-remember maintenance requirements in place that will amount to making the tests flaky.

**Comment by dsotirakis** [comment]
@sdboyer -- tried `mergefs` again, but unfortunately as it says [here](https://github.com/laher/mergefs/blob/master/mergefs.go#L13) `The first FS takes precedence`, so when it comes to combining filesystems only cares about the first FS really (which is the `testdata` in our case) so it makes no difference / can't see root :(. Made it work before, but I think I missed something there. So bad thing that we need the `cue/scuemata` in every different test folder, but I get it as it's a subject to change maybe in the future? (meaning the strict structure we have)

Good thing that we can get rid of the `go:embed` in tests though.

**Review by sdboyer — APPROVED** [review]

**Pull request merged**